### PR TITLE
Add remove

### PR DESCRIPTION
**Pull request by SimonGideon**
> ## :fleur_de_lis: :1234:  To Do list: add & remove
> ## What does this PR do?
> This PR is raised to implement the following features: +1 :
> 
> * [x] Use ES6 syntax..
> * [x] Remove all hardcoded items from the tasks array.
> * [x] Create a new JavaScript file for the new functionality.
> * [x] Implement a function for adding a new task (add a new element to the array).
> * [x] Implement a function for deleting a task (remove an element from the array).
> * [x] Implement a function for editing task descriptions.
> * [x] By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
> * [x] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
> * [x] All changes to the To Do List should be saved in local storage.

> #### Thanks for reviewing my PR :smile:

### CODE REVIEW

File: src/models/components.js [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+
+// editing todos
+const editTask = (task) => {
+  const taskItem = task.children[0].children[1].name;
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  const taskIndex = tasks.findIndex((x) => x.description === taskItem);
+  const taskName = task.querySelector('#task-name').value;
+  tasks[taskIndex].description = taskName;
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
```
Comment (franclobo) [inline src/models/components.js:37]: - [ ] Please 🙏🏽 correct this part of the code, when I tried to edit the task I clicked on it and the cursor appeared and I could change the word, but when I refreshed the page it returned to the previous word, it means that the task is not saving in the local storage. **All changes to the To Do List should be saved in local storage.**

File: src/models/components.js [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
+// function for adding task to list
+const addTask = (task) => {
+  const taskObj = {};
+  taskObj.index = taskArr.length + 1;
+  taskObj.description = task;
+  taskObj.completed = false;
+  displayTask(taskObj);
+  taskArr.push(taskObj);
+  localStorage.setItem('tasks', JSON.stringify(taskArr));
+};
```
Comment (franclobo) [inline src/models/components.js:27]: - [ ] When I added a task it saved in the local storage, but when I add a new one and refreshed the page, the previous task disappeared. Kindly fix it. 👇🏽 **All changes to the To Do List should be saved in local storage.** 

https://user-images.githubusercontent.com/58642949/199861224-dfdbd926-9be8-460d-bbaf-39ab1826ad1d.mp4

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @SimonGideon, 

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️
- Descriptive PR ✅
- Correct Github flow ☑️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please fix the bugs 👇🏽 

image

**Hint:** Declare the mode in the `webpack.config.js file`, like this 👇🏽 

image


- [ ] Please 🙏🏽 add the start attribute `"start": "webpack serve --open,"` in the scripts on pakage.json to open the webpack server. 📂 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/models/components.js (hunk 1 above)
Comment (SimonGideon) [reply src/models/components.js:37]: :100: Thanks for the review. the requested changes has been noted and implemented accordingly

File: src/models/components.js (hunk 2 above)
Comment (SimonGideon) [reply src/models/components.js:27]: Thanks for this the video made the explanation so clear and understandable. Now its fixed and good to go. :smile:

**Review by DammyShittu — APPROVED** [review]
Hi @SimonGideon,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**